kratos/includes/properties.h

```diff
@@ -91,6 +91,9 @@ class Properties : public IndexedObject
 
     typedef std::unordered_map<std::size_t, TableType> TablesContainerType; // This is a provisional implmentation and should be changed to hash. Pooyan.
 
+    typedef class PropertiesAccessor AccessorType;
+
+    typedef Kratos::shared_ptr<AccessorType> AccessorPointerType;
```

Comment:
why do we need a shared pointer here? i don't think we want to move the accessor around (although we may copy it somewhere else).

In short i would pass around references to the accessor but never Pointers

Reply:
It has to be a pointer bcs of the forward declaration
I tried with unique ptrs but didn't get it to work, long story short bcs of some destructor issues (see [here](https://www.fluentcpp.com/2017/09/22/make-pimpl-using-unique_ptr/))
I will talk to Jordi abt it